eqt/ui/UIStackedWidget.py

```diff
@@ -0,0 +1,108 @@
+from eqt.ui.UIFormWidget import UIFormFactory
+from PySide2 import QtWidgets
+from PySide2.QtCore import Qt
+from PySide2.QtWidgets import QHBoxLayout, QListWidget, QStackedWidget, QWidget
+
+
+class UIStackedWidget(object):
+    '''
+             QWidget or QDockWidget
+    +----------------------------------------------------------+
+    |        QHBoxLayout                                       |
+    |   +---------------------------------------------------+  |
+    |   |                                                   |  |
+    |   |    +--------------+   +------------------------+  |  |
+    |   |    |   QList      |   |  QStackedWidget        |  |  |
+    |   |    |              |   |                        |  |  |
+    |   |    |              |   |                        |  |  |
+    |   |    +--------------+   +------------------------+  |  |
+    |   |                                                   |  |
+    |   +---------------------------------------------------+  |
+    |                                                          |
+    +----------------------------------------------------------+
+    '''
+
+    def createStack(self):
+        self.stack_list = QListWidget()
+        self.margin_size = 20
+        self.stack_list.setStyleSheet("margin : {}px".format(self.margin_size))
+
+        self.Stack = QStackedWidget(self)
+
+        hbox = QHBoxLayout(self)
+        hbox.addWidget(self.stack_list)
+        hbox.addWidget(self.Stack)
+        hbox.setAlignment(self.stack_list, Qt.AlignTop)
+
+        self.setLayout(hbox)
+        self.stack_list.currentRowChanged.connect(self.display)
+
+        self.tabs = {}
+
+        self.num_tabs = 0
+        self.widgets = {}
+
+    def display(self, i):
+        self.Stack.setCurrentIndex(i)
+
+    def currentIndex(self):
+        return self.Stack.currentIndex()
+
+    def addTab(self, title, widget='form'):
+        self.stack_list.addItem(title)
+
+        if widget == 'form':
+            widget = UIFormFactory.getQWidget(self)
+
+        self.Stack.addWidget(widget)
+        self.tabs[title] = widget
+        self.num_tabs += 1
+        height_multiplier = self.num_tabs + 1.3
```

Comment:
There is something a bit strange going on here, with the set height of the QListWidget. I need to spend some more time looking at this, but for now it seems to work